docs/using-the-jdbc-driver/using-plugins/UsingTheReadWriteSplittingPlugin.md

```diff
@@ -113,3 +113,14 @@ There are many session state attributes that can change during a session, and ma
 All other session state attributes will be lost when switching connections between the writer/reader.
 
 If your SQL workflow depends on session state attributes that are not mentioned above, you will need to re-configure those attributes each time that you switch between the writer/reader.
+
+
+### Spring limitations
+
+#### @Transactional(readOnly = True)
+
+We recommend that you do not use the read/write splitting plugin in conjunction with Spring's @Transactional(readOnly = True) annotation. When a method with this annotation is hit, Spring calls conn.setReadOnly(true), executes the method, and then calls setReadOnly(false) to restore the connection's initial readOnly value. Consequently, every time the method is called, the plugin switches to the reader, executes the method, and then switches back to the writer. Although the reader connection will be cached after the first setReadOnly call, there is still some overhead when switching between the cached writer/reader connections. This constant switching is not an ideal use of the plugin because it is frequently incurring this overhead. The suggested approach for this scenario is to avoid loading the read/write splitting plugin and instead use the writer cluster URL for your write operations and the reader cluster URL for your read operations. By doing this you avoid the overhead of constantly switching between connections while still spreading load across the database instances in your cluster. 
```

Comment:
WARNING: The use of RWSplitting with the annotation @Transactional(readOnly = True) is not recommended.

When...